### ISSUE TITLE
converting ld's functions to attributes

### ISSUE THREAD

**Issue by nir0s** [nir0s/ld#101]
It would be nice (while preserving backward compatibility of course) to be able to ask for `ld.id` instead of `ld.id()` and `ld.pretty_version` instead of `ld.version(pretty=True)`.

**Comment by andy-maier** [comment]
With the package name change going on in parallel, I think we can afford to break compatibility on this as well, as users need to touch their code anyway.

I first thought why not, but then I found a few disadvantages of introducing properties:

* For a property, it is not possible to extend its functionality like one can do for a function or method by adding an argument with a default value. We did that with the `best` argument in versions, for example.
* The `version()` function currently has `pretty` and `best` boolean arguments, and converting that function into properties means to multiply out all the four combinations. The end result is not necessarily more intuitive than the function. But that is the only one with two arguments.
* If we end up having some attributes as functions (for the reasons above) and some as properties, one needs to remember which attributes are available in which way.
* If we end up having both a property and a function for some attribute, they will have to have different names, and we need to make sure to document that in such a way that readers understand they provide the same information.

I'm not saying we should not go there. If you think we should go there, can you propose which functions become properties, and which stay functions, and for which we have both (if at all)?

**Comment by nir0s** [comment]
I suggested this because it seems more intuitive to me. You raise reasonable concerns.

* Extending the functionality of each of our functions is possible, but you're right, it will not be as simple as it is now.
* I'm aware of the fact that we'll have to provide additional properties for each function due to arguments. I'm not certain it's worth it. It's just a thought.
* Regarding the two last issues, I think we should either have all properties or all functions. I think it's a bad idea to have both except for `linux_distribtution` (as the reason we implemented it in the first place is to provide a similar experience to the current stdlib implementation.

Honestly, with the issues you're raising here, i'm not at all sure it's a good idea to convert to properties. It's not that important and I don't think this will be a blocker in any way to anyone.
